src/query/plan/rewrite/edge_type_index_lookup.hpp

```diff
@@ -599,19 +633,18 @@ class EdgeTypeIndexRewriter final : public HierarchicalLogicalOperatorVisitor {
     }
 
     if (EdgeTypePropertyIndexingPossible()) {
-      // This is currently the simplest way the get the most out of edge-type + proeprty indexing.
-      // We only remove the filter if we are looking for edges with a specific edge-type. If we
-      // have some other type of property filtering e.g. range or value lookup, we do not want to
-      // remove the filters to keep the semantics of the original query intact. We can optimize
-      // this by adding additional operators for the specific kinds of edge-type+property filtering.
-      // On top of this we only want to remove the NOT NULL filter
       const auto prop_filter_type = property_filter_->property_filter->type_;
-      if (prop_filter_type == PropertyFilter::Type::IS_NOT_NULL) {
-        filter_exprs_for_removal_.insert(property_filter_->expression);
-        filters_.EraseFilter(*property_filter_);
+      switch (prop_filter_type) {
+        case PropertyFilter::Type::IS_NOT_NULL:
+        case PropertyFilter::Type::EQUAL: {
+          filter_exprs_for_removal_.insert(property_filter_->expression);
+          filters_.EraseFilter(*property_filter_);
+          return std::make_unique<ScanAllByEdgeTypeProperty>(input, output_symbol, edge_type_, *property_, view);
+        }
+        default: {
+          spdlog::warn("Filter is not supported by edge type property index.");
+        }
```

Comment:
I don't think `spdlog::warn` is necessary here. The user can't do anything about it, so I would rather remove it.

Reply:
Thank you!